src/solve.jl

```diff
@@ -51,19 +51,19 @@ The callback function `callback` is a function which is called after every optim
 step. Its signature is:
 
 ```julia
-callback = (u, loss_val, other_args) -> false
+callback = (u, loss_val, other_args...) -> false
 ```
 
 where `u` and `loss_val` are the current optimization variables and loss/objective value
 in the optimization loop and `other_args` can be the extra things returned from
-the optimization `f`. This allows for saving values from the optimization and
-using them for plotting and display without recalculating. The callback should
-return a Boolean value, and the default should be `false`, such that the optimization
-gets stopped if it returns `true`.
+the optimization `f` or the state of the optimizer and iteration count. This allows 
```

Comment:
" or the state of the optimizer and iteration count" what do you mean by that?

Reply:
I'll refine it. It's a placeholder to make sure I update it as soon as the update happens in the wrappers.

The point is that the callback will have a current state of the optimizer if it's available and iteration counts

Reply:
I don't understand how that would work. Wouldn't that need to be an argument? If so, how is that argument uniformized? I don't see a struct definition or docs to go along with that.